src/validation.cpp

```diff
@@ -2359,6 +2359,78 @@ CScript calculate_contract(const CScript& federationRedeemScript, const CScript&
     return scriptDestination;
 }
 
+bool GetAmountFromParentChainPegin(CAmount& amount, const Sidechain::Bitcoin::CTransaction& txBTC, unsigned int nOut)
+{
+    amount = txBTC.vout[nOut].nValue;
+    return true;
+}
+
+template<typename T>
```

Comment:
I'm curious about the use of a template here (vs., say, an inline static function).  This isn't a header file and there only appears to be one use of this template/function -- is this a project coding convention or is there some performance/maintenance gain?

Reply:
I'm not the biggest fan of adding code for future use into a pull request, it could sit there making no sense for a long time.

Reply:
Yeah, I did it only because @instagibbs asked. Perhaps I should have done 2 commits but not 2 PRs